src/components/Purchase/CheckoutForm.js

```diff
@@ -0,0 +1,116 @@
+import React, { useEffect, useState } from 'react';
+import {
+  PaymentElement,
+  LinkAuthenticationElement,
+  useStripe,
+  useElements,
+} from '@stripe/react-stripe-js';
+import style from './CheckoutForm.module.css';
+
+export default function CheckoutForm({ paymentAmount }) {
+  const stripe = useStripe();
+  const elements = useElements();
+
+  const [email, setEmail] = useState('');
+  const [message, setMessage] = useState(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isStripeLoaded, setIsStripeLoaded] = useState(false);
+  useEffect(() => {
+    if (!stripe) {
+      return;
+    }
+    // Stripe.js has loaded
+    setIsStripeLoaded(true);
+    const clientSecret = new URLSearchParams(window.location.search).get(
+      'payment_intent_client_secret'
+    );
+
+    if (!clientSecret) {
+      return;
+    }
+
+    stripe.retrievePaymentIntent(clientSecret).then(({ paymentIntent }) => {
+      switch (paymentIntent.status) {
+        case 'succeeded':
+          setMessage('Payment succeeded!');
+          break;
+        case 'processing':
+          setMessage('Your payment is processing.');
+          break;
+        case 'requires_payment_method':
+          setMessage('Your payment was not successful, please try again.');
+          break;
+        default:
+          setMessage('Something went wrong.');
+          break;
+      }
+    });
+  }, [stripe]);
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    if (!isStripeLoaded || !stripe || !elements) {
+      // Stripe.js hasn't yet loaded.
+      return;
+    }
+
+    setIsLoading(true);
+
+    const { error } = await stripe.confirmPayment({
+      elements,
+      confirmParams: {
+        return_url: `${window.location.origin}/checkout/completed`,
+      },
+    });
+
+    // This point will only be reached if there is an immediate error when
+    // confirming the payment. Otherwise, customer will be redirected to
+    // `return_url`. For some payment methods like iDEAL, your customer will
+    // be redirected to an intermediate site first to authorize the payment, then
+    // redirected to the `return_url`.
+    if (error.type === 'card_error' || error.type === 'validation_error') {
+      setMessage(error.message);
+    } else {
+      setMessage('An unexpected error occurred.');
+    }
+
+    setIsLoading(false);
+  };
+
+  const paymentElementOptions = {
+    layout: 'tabs',
+  };
+
+  return (
+    <form
+      id="payment-form"
+      onSubmit={handleSubmit}
+      className={style.checkoutFormContainer}
+    >
+      <p>
+        Your payment amount: <b>${paymentAmount / 100}</b>
+      </p>
+      <LinkAuthenticationElement
+        id="link-authentication-element"
+        onChange={(e) => setEmail(e.target.value)}
+      />
```

Comment:
This component's `onChange` doesn't return a normal event object with a target field. So this is currently causing some errors that you can see in the console when you type into the input box generated by this element:

![](https://github.com/Code-the-Dream-School/dd-prac-team4-front/assets/9094098/8b0d8b3e-5a2a-4680-809c-1bb3339b1435)

If you set a breakpoint on this line, you'll see that the shape of the argument `e` is actually a bit different than we expect, and the email is actually under `e.value` not `e.target.value`:

![](https://github.com/Code-the-Dream-School/dd-prac-team4-front/assets/9094098/3ab9cd99-f1e2-4954-876d-c9311bfd29d2)

---

for bonus insight, if we dig into this component's props in the react-stripe-js library, we can see they're defined here in a typescript file:

[https://github.com/stripe/react-stripe-js/blob/b516a5e44c03de91093162acf1b33d89a7fbc446/src/types/index.ts#L334-L340](https://github.com/stripe/react-stripe-js/blob/b516a5e44c03de91093162acf1b33d89a7fbc446/src/types/index.ts#L334-L340)

which is then further defiend here in the stripe-js library:

[https://github.com/stripe/stripe-js/blob/4ba4eadc8f14040f9dc0bf916b9c09c383f6d603/types/stripe-js/elements/link-authentication.d.ts#L136-L158](https://github.com/stripe/stripe-js/blob/4ba4eadc8f14040f9dc0bf916b9c09c383f6d603/types/stripe-js/elements/link-authentication.d.ts#L136-L158)

Which matches the fields we saw when we did the breakpoint

Reply:
Thank you! Good point! I fixed this!